agent/handlers/credentials/handler.go

```diff
@@ -81,9 +96,10 @@ func ServeHttp(credentialsManager credentials.Manager, containerInstanceArn stri
 }
 
 // setupServer starts the HTTP server for serving IAM Role Credentials for Tasks.
-func setupServer(credentialsManager credentials.Manager, auditLogger audit.AuditLogger) http.Server {
+func setupServer(credentialsManager credentials.Manager, auditLogger audit.AuditLogger) *http.Server {
 	serverMux := http.NewServeMux()
-	serverMux.HandleFunc(credentials.CredentialsPath, credentialsV1RequestHandler(credentialsManager, auditLogger))
+	serverMux.HandleFunc(credentials.V1CredentialsPath, credentialsV1V2RequestHandler(credentialsManager, auditLogger, getV1CredentialsID, apiVersion1))
+	serverMux.HandleFunc(credentials.V2CredentialsPath+"/", credentialsV1V2RequestHandler(credentialsManager, auditLogger, getV2CredentialsID, apiVersion2))
```

Comment:
Are requests of the format `/v2/credentials/id/blah` handled by the mux library of do they get handled by `credentialsV1V2RequestHandler`? I'm just curious as to the error code returned (`302` or `4xx`).